File: patch_strd.js
const fs = require('fs');
const files = ['pacs9/pacs9.component', 'pacs9adv/pacs9adv.component', 'pacs9cov/pacs9cov.component'];
const base = 'c:/Users/HP/Desktop/iso final/iso20022generatorfrontend/src/app/pages/manual-entry/pacs';

for (const f of files) {
    const htmlPath = base + '/' + f + '.html';
    if (fs.existsSync(htmlPath)) {
        let html = fs.readFileSync(htmlPath, 'utf8');
        html = html.replace(/<option value="strd">Structured Remittance<\/option>/g, '');
        
        const startStr = '<ng-container *ngIf="form.get(\'rmtInfType\')?.value === \'strd\'">';
        const startIndex = html.indexOf(startStr);
        if (startIndex >= 0) {
            const endIndex = html.indexOf('</ng-container>', startIndex);
            if (endIndex >= 0) {
                html = html.substring(0, startIndex) + html.substring(endIndex + '</ng-container>'.length);
            }
        }
        fs.writeFileSync(htmlPath, html, 'utf8');
    }

    const tsPath = base + '/' + f + '.ts';
    if (fs.existsSync(tsPath)) {
        let ts = fs.readFileSync(tsPath, 'utf8');
        
        ts = ts.replace(/^\s*rmtInfStrdCdtrRefType:.*$/gm, '');
        ts = ts.replace(/^\s*rmtInfStrdCdtrRef:.*$/gm, '');
        ts = ts.replace(/^\s*rmtInfStrdAddtlRmtInf:.*$/gm, '');
        
        ts = ts.replace(/\} else if \(rmtInfType\?\.value === 'strd'\) \{[^\}]+?\}/g, '}');
        
        ts = ts.replace(/\} else if \(this\.form\.get\('rmtInfType'\)\?\.value === 'strd'\) \{[\s\S]*?\}(?=\s*(?:\} else|if|return|\}))/, '}');
        
        fs.writeFileSync(tsPath, ts, 'utf8');
    }
}

File: patch_strd.py
import os
import re

files = [
    r'c:\Users\HP\Desktop\iso final\iso20022generatorfrontend\src\app\pages\manual-entry\pacs\pacs9\pacs9.component.ts',
    r'c:\Users\HP\Desktop\iso final\iso20022generatorfrontend\src\app\pages\manual-entry\pacs\pacs9adv\pacs9adv.component.ts',
    r'c:\Users\HP\Desktop\iso final\iso20022generatorfrontend\src\app\pages\manual-entry\pacs\pacs9cov\pacs9cov.component.ts'
]

for f in files:
    if os.path.exists(f):
        with open(f, 'r', encoding='utf-8') as file:
            content = file.read()
        
        # Remove buildForm fields
        content = re.sub(r'\s*rmtInfStrdRfrdDocNb: \[''.*?\],', '', content)
        content = re.sub(r'\s*rmtInfStrdRfrdDocCd: \[''.*?\],', '', content)
        content = re.sub(r'\s*rmtInfStrdRfrdDocAmt: \[''.*?\].*?,', '', content)
        content = re.sub(r'\s*rmtInfStrdRfrdDocAmt: \[''.*?\]\]', '', content)
        content = re.sub(r'\s*rmtInfStrdCdtrRefType: \[''.*?\],', '', content)
        content = re.sub(r'\s*rmtInfStrdCdtrRef: \[''.*?\],', '', content)
        content = re.sub(r'\s*rmtInfStrdAddtlRmtInf: \[''.*?\],', '', content)
        
        # Remove conditional validation block
        content = re.sub(r'\} else if \(rmtInfType\?\.value === \'strd\'\) \{.*?\}\s*const ctry', '} const ctry', content, flags=re.DOTALL)
        content = re.sub(r'\} else if \(this\.form\.get\(\'rmtInfType\'\)\?\.value === \'strd\'\) \{.*?\}\s*const ctry', '} const ctry', content, flags=re.DOTALL)
        content = re.sub(r'\} else if \(rmtInfType\?\.value === \'strd\'\) \{.*?\}\s*const chrg', '} const chrg', content, flags=re.DOTALL)
        
        # Remove XML generator block
        content = re.sub(r'\} else if \(v\.rmtInfType === \'strd\'\) \{.*?\}\s*const tval', '} const tval', content, flags=re.DOTALL)
        
        # Remove XML parser block
        content = re.sub(r'\} else \{\s*const strd = getT\(\'Strd\', rmtInf\);.*?\}\s*\}\s*\} else \{', '} } else {', content, flags=re.DOTALL)
        
        with open(f, 'w', encoding='utf-8') as file:
            file.write(content)
